app/assets/javascripts/admin/addon/lib/setting-object-helper.js

```diff
@@ -0,0 +1,76 @@
+import { computed } from "@ember/object";
+import { readOnly } from "@ember/object/computed";
+import { isPresent } from "@ember/utils";
+import { deepEqual } from "discourse-common/lib/object";
+import { i18n } from "discourse-i18n";
+
+export default class SettingObjectHelper {
+  @readOnly("settingObj.allow_any") anyValue;
+
+  constructor(settingObj) {
+    this.settingObj = settingObj;
+  }
+
+  @computed("settingObj.value", "settingObj.default")
```

Comment:
Using legacy computed properties in new code is not ideal. Can we drop them in this class?

We may need to do `this.settingObj.get("...")` to access EmberObject properties in an autotracking way. And we can add `@dependentKeyCompat` if these getters need to be accessed **from** a legacy context.

(useful ref: internal `t/125342`